src/MartinGeorgiev/Doctrine/DBAL/Types/Point.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MartinGeorgiev\Doctrine\DBAL\Types;
+
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use MartinGeorgiev\Doctrine\DBAL\Types\Exceptions\InvalidPointForDatabaseException;
+use MartinGeorgiev\Doctrine\DBAL\Types\Exceptions\InvalidPointForPHPException;
+use MartinGeorgiev\Doctrine\DBAL\Types\ValueObject\Point as PointValueObject;
+
+/**
+ * Implementation of PostgreSQL POINT data type.
+ *
+ * @see https://www.postgresql.org/docs/17/datatype-geometric.html#DATATYPE-GEOMETRIC-POINTS
+ * @since 3.1
+ *
+ * @author Sébastien Jean <sebastien.jean76@gmail.com>
+ */
+class Point extends BaseType
+{
+    protected const TYPE_NAME = 'point';
+
+    public function convertToDatabaseValue($value, AbstractPlatform $platform): ?string
+    {
+        if ($value === null) {
+            return null;
+        }
+
+        if (!$value instanceof PointValueObject) {
+            throw InvalidPointForPHPException::forInvalidType($value);
+        }
+
+        return (string) $value;
+    }
+
+    public function convertToPHPValue($value, AbstractPlatform $platform): ?PointValueObject
+    {
+        if ($value === null) {
+            return null;
+        }
+
+        if (!\is_string($value)) {
+            throw InvalidPointForDatabaseException::forInvalidType($value);
+        }
+
+        if (!\preg_match('/\((-?\d+(?:\.\d{1,6})?),\s*(-?\d+(?:\.\d{1,6})?)\)/', $value, $matches)) {
```

Comment:
Given the repetitiveness of this expression in the current class and in `PointArray`, it made me think that it also applies to the construction of `Point` VO. We don't really expect invalid/unpersistable VO instances to be in the application, right?

Reply:
Maybe `\((-?\d+(?:\.\d{1,6})?)` can be a public constant in the `Point` VO and all other places will reference this accordingly for ease of maintenance?

Reply:
> Maybe `\((-?\d+(?:\.\d{1,6})?)` can be a public constant in the `Point` VO and all other places will reference this accordingly for ease of maintenance?

I added a public constant.

Reply:
Thank you :)